src/main/java/com/example/onlinebookstore/service/BookServiceImpl.java

```diff
@@ -1,23 +1,39 @@
 package com.example.onlinebookstore.service;
 
-import com.example.onlinebookstore.model.Book;
+import com.example.onlinebookstore.dto.BookDto;
+import com.example.onlinebookstore.dto.BookRequestDto;
+import com.example.onlinebookstore.exceptions.EntityNotFoundException;
+import com.example.onlinebookstore.mapper.BookMapper;
 import com.example.onlinebookstore.repository.BookRepository;
 import java.util.List;
-import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
-@RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    public BookServiceImpl(BookRepository bookRepository, BookMapper bookMapper) {
+        this.bookRepository = bookRepository;
+        this.bookMapper = bookMapper;
+    }
+
+    @Override
+    public BookDto save(BookRequestDto bookRequestDto) {
+        return bookMapper.toDto(bookRepository.save(bookMapper.toModel(bookRequestDto)));
+    }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public BookDto getById(Long id) {
+        return bookMapper.toDto(bookRepository.findById(id)
+                .orElseThrow(() ->
+                        new EntityNotFoundException("Can't find book by this id: " + id)));
```

Comment:
you should return Optional from the repository level, when you update it - don't use ifPresent here

there is a map method, so that you can do everything in a single chain